src/bpf/bpfd-manager.c

```diff
@@ -0,0 +1,489 @@
+                        if (r < 0) {
+                                log_error_errno(r, "Failed to build JSON object: %m");
+                                continue;
+                        }
+
+                        /* Check if this is the final bpf program. If so, send a varlink reply instead of notify */
+                        if (sym_bpf_prog_get_next_id(id, &next_id) < 0) {
+                                r = varlink_reply(link, v);
+                                if (r < 0) {
+                                        log_error_errno(r, "Failed to varlink reply: %m");
+                                        continue;
+                                }
+                        } else {
+                                r = varlink_notify(link, v);
+                                if (r < 0) {
+                                        log_error_errno(r, "Failed to varlink notify: %m");
+                                        continue;
+                                }
+                        }
+                } else {
+                        log_struct(LOG_INFO,
+                                LOG_MESSAGE("Logging info about BPF program: %s", prog_name),
+                                "MESSAGE_ID=" SD_MESSAGE_BPFD_LOG_STR,
+                                "NAME=%s", prog_name,
+                                "TYPE=%u", prog_info.type,
+                                "MEMLOCK=%s", memlock,
+                                "TAG=%s", prog_tag,
+                                "MAP_NAMES=%s", prog_map_names,
+                                "RUN_TIME_NS=%llu", prog_info.run_time_ns,
+                                "RUN_COUNT=%llu", prog_info.run_cnt);
+                }
+        }
+
+        char *map_name;
+        HASHMAP_FOREACH(map_name, map_name_by_id) {
+                free(map_name);
+        }
+        return 0;
+}
+
+static int log_bpf_maps_and_progs(Manager *m) {
+        int r = fetch_bpf_maps_and_progs(NULL);
+        if (r < 0)
+                return r;
+        return 0;
+}
+
+static int vl_method_get_bpf_progs_and_maps(Varlink *link, JsonVariant *parameters, VarlinkMethodFlags flags, void *userdata) {
+        int r;
+
+        r = fetch_bpf_maps_and_progs(link);
+        if (r < 0)
+                return r;
+
+        return 0;
+}
+
+static int manager_bind_varlink(Manager *m) {
+        const char *socket_path;
+        int r;
+
+        assert(m);
+        assert(!m->varlink_server);
+
+        r = varlink_server_new(&m->varlink_server, VARLINK_SERVER_INHERIT_USERDATA);
+        if (r < 0)
+                return log_error_errno(r, "Failed to allocate varlink server object: %m");
+
+        varlink_server_set_userdata(m->varlink_server, m);
+
+        r = varlink_server_bind_method(
+                        m->varlink_server,
+                        "io.systemd.BpfProgsAndMaps.GetBpfProgsAndMaps", vl_method_get_bpf_progs_and_maps
```

Comment:
We should probably start with a generic `Describe` method so we can add more stuff here later if we want. If needed we can add more specific methods as well that only return some specific type of data later.